suprb/optimizer/rule/es/es.py

```diff
@@ -99,6 +100,39 @@ def _optimize(self, X: np.ndarray, y: np.ndarray, initial_rule: Rule, random_sta
                 warnings.warn("No valid children were generated during this iteration.", UserWarning)
                 continue
 
+            # For CSR only the spread mutation rate gets altered, for MPR both rate are altered
+            if self.adaptive_sigma:
+                matching_type = elitist.match.__class__.__name__
+                fitter_children = sum([1 for child in children if child.fitness_ > elitist.fitness_])
+                proportion = fitter_children / self.lmbda
+
+                if proportion > 0.2:
```

Comment:
you could easily move the duplicated block to a function that takes as input the factor (.85 or 2) and the operator (/ or *) removing a lot of that code duplication